thunder/tests/opinfos.py

```diff
@@ -2531,6 +2531,67 @@ def where_error_generator(op, device, dtype=torch.float32, **kwargs):
 conditional_and_mask_ops.append(where_opinfo)
 
 
+def nan_to_num_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    a = make((4, 4), dtype=dtype, requires_grad=requires_grad)
+    double_max = torch.finfo(torch.float64).max
+    double_min = torch.finfo(torch.float64).max
+    if dtype == torch.FloatType:
```

Comment:
These checks aren't quite write. This wants to check if the dtype is a float dtype or a complex dtype. The value of `dtype` will be something like `torch.float32`, which will not compare equal to `torch.FloatType`. Instead, this can query the `is_floating_point` and `is_complex` attributes of PyTorch dtypes, like this:

```
if dtype.is_floating_point:
  ...
elif dtype.is_complex:
  ...
```

Reply:
I completely missed that, thanks so much!